assets/src/edit-story/components/library/panes/media/deleteDialog.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { useCallback } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../../../app/api';
+import { Plain } from '../../../../components/button';
+import Dialog from '../../../../components/dialog';
+import { useSnackbar } from '../../../../app/snackbar';
+import { useMedia } from '../../../../app/media';
+
+const DialogBody = styled.div`
+  width: 512px;
+  font-family: ${({ theme }) => theme.fonts.body1.family};
+  line-height: ${({ theme }) => theme.fonts.body1.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.body1.weight};
+  font-size: ${({ theme }) => theme.fonts.body1.size};
+  color: ${({ theme }) => theme.colors.fg.v0};
+`;
+
+/**
+ * Display a confirmation dialog for when a user wants to delete a media element.
+ *
+ * @param {Object} props Component props.
+ * @param {number} props.mediaId Selected media element.
+ * @param {function()} props.onClose Callback to toggle dialog display on close.
+ * @return {null|*} The dialog element.
+ */
+function DeleteDialog({ mediaId, onClose }) {
+  const {
+    actions: { deleteMedia },
+  } = useAPI();
+  const { showSnackbar } = useSnackbar();
+  const {
+    actions: { deleteMediaElement },
+  } = useMedia();
+
+  const onDelete = useCallback(async () => {
+    onClose();
```

Comment:
Should this be at the end of this async function?

Reply:
I put this here so it'll be harder for the user to accidentally try to delete the same item twice. (However the user might get confused if the item is still in the media gallery).

If we do put it in the finally clause, we'd have to disable the delete button while the delete operation is ongoing. I'm just a bit worried that it might be seen as "slow", so if the user wants to delete multiple items they're "stuck" on the dialog screen in between each delete. 

I don't feel strongly about either approaches, WDYT?